### PR TITLE
Refactor chain-selector to use chain specific chainID

### PR DESCRIPTION
**Pull request by huangzhen1997**
https://smartcontract-it.atlassian.net/browse/NONEVM-827

- To generalize plugin [component](https://github.com/smartcontractkit/chainlink/blob/cf726dbc714b7695d3fd98649d35f0483d81f961/core/capabilities/ccip/oraclecreator/plugin.go#L130) to support Solana, we need to help CCIP team to refactor chain-selector. The existing chainID is uint64, this PR converts it to string since for solana we will use genesis hash as chainID.

- Instead of use previous selector.yml (`chainID`-> `{selector+name}`) and selector_families(`selector` -> `family`), provides a reconstructed config using `selector` -> `{chainID + name + family}`, and making chainID a string instead of uint64.

- update rest of the code/test to use the `selector_restructured.yml` config that key by selector. The original `selectors.yml` and `selector_families.yml` are left for reference and can be removed in a following PR.

- preserve the existing functions that use uint64 chainID for backward compatibility, but mark as deprecated

### CODE REVIEW

File: selectors.go [hunk 1]
```diff
@@ -32,153 +29,215 @@ const (
...
+		if v.Family == "" {
+			continue
+		}
+
+		// update chainIDToSelectorMapForFamily
+		_, exist := chainIDToSelectorMapForFamily[v.Family]
+		if exist {
+			chainIDToSelectorMapForFamily[v.Family][v.ChainID] = k
+		} else {
+			chainIDToSelectorMapForFamily[v.Family] = make(map[string]uint64)
```
Comment (prashantkumar1982) [inline selectors.go:86]: You need to also call the below for this case right:
chainIDToSelectorMapForFamily[v.Family][v.ChainID] = k

Comment (huangzhen1997) [reply selectors.go:84]: Oh yes, thanks

File: selectors.go [hunk 2]
```diff
@@ -32,153 +29,215 @@ const (
...
 	}
-	return 0, fmt.Errorf("chain selector not found for chain %d", chainId)
+
+	return selector, nil
 }
 
-func NameFromChainId(chainId uint64) (string, error) {
-	details, exist := evmChainIdToChainSelector[chainId]
+func NameFromChainIdAndFamily(chainId string, family string) (string, error) {
+	// if family is missing use EVM as default
```
Comment (prashantkumar1982) [inline selectors.go:161]: Instead of duplicating all this code from above function, just call that?
SelectorFromChainIdAndFamily()
Then from selector, just fetch details.

File: selectors.go [hunk 3]
```diff
@@ -32,153 +29,215 @@ const (
...
+		family = FamilyEVM
 	}
-	chainId, err := strconv.ParseUint(name, 10, 64)
-	if err == nil {
-		if details, exist := evmChainIdToChainSelector[chainId]; exist && details.ChainName == "" {
-			return chainId, nil
+
+	for _, v := range selectorToChainDetails {
+		if v.Name == name && family == v.Family {
+			return v.ChainID, nil
```
Comment (prashantkumar1982) [inline selectors.go:196]: Has our schema enforced anywhere that name, family paid must be unique?
I know practically we use unique names, but nothing currently prevents from someone naming say 2 different EVM chains as same?

Just wondering, if we leave that logic as it-is, or add some validation?

Comment (prashantkumar1982) [reply selectors.go:196]: I guess we could add a unit-test to ensure that name+family combination should not have duplicates, if others agree with this.

File: selectors.go [hunk 4]
```diff
@@ -32,153 +29,187 @@ const (
...
+		if v.Family == "" {
+			continue
+		}
+
+		// update testChainIDToSelectorMapForFamily
+		_, exist := testChainIDToSelectorMapForFamily[v.Family]
+		if exist {
+			testChainIDToSelectorMapForFamily[v.Family][v.ChainID] = k
+		} else {
+			testChainIDToSelectorMapForFamily[v.Family] = make(map[string]uint64)
```
Comment (prashantkumar1982) [inline selectors.go:58]: This too

Comment (huangzhen1997) [reply selectors.go:58]: thanks, I am actually going to combine regular and test chains, noticed `var evmChainIdToChainSelector = loadAllSelectors()` has the two combined

**Comment by andrevmatos** [comment]
> The original selectors.yml and selector_families.yml are left for reference and can be removed in a following PR.

Be careful, this project is public, and many internal and external dependents require exact format, and even hardcode the selectors.yml URL from this repo. Any such change should be a major version bump, and we'd be better doing as much as we can to keep backwards compatibility on this.

**Comment by huangzhen1997** [comment]
> > The original selectors.yml and selector_families.yml are left for reference and can be removed in a following PR.
> 
> Be careful, this project is public, and many internal and external dependents require exact format, and even hardcode the selectors.yml URL from this repo. Any such change should be a major version bump, and we'd be better doing as much as we can to keep backwards compatibility on this.

I kept function like `SelectorFromChainId()` and `ChainIdFromName()` for backward compatibility

**Review by makramkd — CHANGES_REQUESTED** [review]
Lets cut a major version - create a v2 directory and place all these changes there?

**Comment by huangzhen1997** [comment]
Replaced with https://github.com/smartcontractkit/chain-selectors/pull/70
